crypto/src/proofs/transparent.rs

```diff
@@ -397,39 +368,27 @@ impl SwapClaimProof {
             .map_err(|_| anyhow!("merkle root mismatch"))?;
 
         // Check that the provided note commitment is for the proof's Swap NFT.
-        let swap_nft_value = Value {
-            amount: 1u64.into(),
-            asset_id: self.swap_nft_asset_id,
-        };
-        let transmission_key_s = self.claim_address.transmission_key_s();
-        let note_commitment_test = note::commitment(
+        gadgets::note_commitment_integrity(
             self.note_blinding,
-            swap_nft_value.clone(),
-            *self.claim_address.diversified_generator(),
-            *transmission_key_s,
-            self.claim_address.clue_key(),
-        );
-
-        if self.note_commitment_proof.commitment() != note_commitment_test {
-            return Err(anyhow!("note commitment mismatch"));
-        }
-
-        // check the swap NFT Asset ID is properly constructed
-        let asset_id = self.swap_nft_asset_id;
-        let expected_plaintext = SwapPlaintext::from_parts(
+            Value {
+                amount: 1u64.into(),
+                asset_id: self.swap_nft_asset_id,
+            },
+            self.claim_address.diversified_generator().clone(),
+            self.claim_address.transmission_key_s().clone(),
+            self.claim_address.clue_key().clone(),
+            self.note_commitment_proof.commitment(),
+        )?;
+
+        // Check that the Swap NFT asset ID is properly constructed.
+        gadgets::asset_id_integrity(
```

Comment:
Could this be `swap_nft_asset_id_integrity` instead, since it's only for swap NFTs and not general asset IDs? Or, we could factor the name into a module path, like `gadgets::swap::asset_id_integrity`?